jumpscale/packages/vdc/billing.py

```diff
@@ -0,0 +1,143 @@
+import os
+import requests
+from decimal import Decimal
+from jumpscale.loader import j
+from jumpscale.sals.vdc.size import VDC_SIZE
+from jumpscale.sals.vdc.models import KubernetesRole
+
+
+BASE_CAPACITY = int(os.getenv("BASE_CAPACITY", 14))
+
+
+def get_addons(flavor, kubernetes_addons):
+    """Get all the addons on the basic user plan
+
+    Args:
+        flavor(str): user flavor for the plan
+        kubernetes_addons(list): all kubernetes nodes 
+    
+    Returns:
+        addons(list): list of addons used by the user over the basic chosen plan
+
+    """
+    plan = VDC_SIZE.VDC_FLAVORS.get(flavor)
+    plan_nodes_count = plan.get("k8s").get("no_nodes")
+    plan_nodes_size = plan.get("k8s").get("size")
+    addons = list()
+    for addon in kubernetes_addons:
+        if addon.role != KubernetesRole.MASTER:
+            if addon.size == plan_nodes_size:
+                plan_nodes_count -= 1
+                if plan_nodes_count < 0:
+                    addons.append(addon)
+            else:
+                addons.append(addon)
+    return addons
+
+
+def get_prices():
+    """Get the prices of the addons and plans from github 
+    Returns:
+        VDC_PRICES(dict): return VDC_prices dictionary form out config
+    """
+    # Store prices from github
+    if not j.config.get("VDC_PRICES"):
+        prices = requests.get(
+            "https://raw.githubusercontent.com/threefoldfoundation/vdc_pricing/master/prices.json"
+        ).json()
+        j.config.set("VDC_PRICES", prices)
+    return j.config.get("VDC_PRICES")
+
+
+def calculate_addon_price(addon):
+    """Calculate the price of a single addon
+    Args:
+        addon(obj): addon object 
+    Returns:
+        (float): addon price
+    """
+    prices = get_prices()
+    size = addon.size.name.lower()
+    return prices["nodes"][size]
+
+
+def calculate_plan_base_price():
+    """Get user plan price
+    Returns:
+        float(): the price of user plan
+    """
+    vdc_instance_name = list(j.sals.vdc.list_all())[0]
+    vdc_instance = j.sals.vdc.get(vdc_instance_name)
```

Comment:
I think we can use `find` with `load_info=True`

Reply:
it didn't work properly 
![image](https://user-images.githubusercontent.com/46320202/101606627-3ea5bd80-3a0c-11eb-83a0-454e3235482c.png)

In all cases, we need it to fail if the VDC instance doesn't exist

Reply:
can you make it `name=` not `vdc_name=`

yes get will create it if not available do we need that ? find won't create it

```
JS-NG> j.sals.vdc.list_all()                                                                          
{'vdc_monday8_waleedhammam'}

JS-NG> j.sals.vdc.find("vdc_monday8_waleedhammam", load_info=True)                                    
UserVDC(
  instance_name='vdc_monday8_waleedhammam',
  vdc_name='monday8',
  owner_tname='waleedhammam',
  solution_uuid='dc5b4209af084dd8be10b463413f6178',
  identity_tid=1721,
  flavor=<VDCFlavor.SILVER: 'silver'>,
  s3=S3(
  parent=UserVDC(instance_name_='vdc_monday8_waleedhammam'),
  minio=S3Container(
  ip_address='0.0.0.0',
  wid=0,
  node_id=None,
  pool_id=0
),
  zdbs=[],
  domain=None
),
  kubernetes=[],
  threebot=VDCThreebot(
  parent=UserVDC(instance_name_='vdc_monday8_waleedhammam'),
  domain=None,
  ip_address='0.0.0.0',
  wid=0,
  node_id=None,
  pool_id=0
),
  created=datetime.datetime(2020, 12, 7, 14, 2, 12, tzinfo=tzutc()),
  last_updated=datetime.datetime(2020, 12, 7, 14, 2, 12, tzinfo=tzutc()),
  expiration=datetime.datetime(2021, 1, 6, 14, 2, 12, tzinfo=tzutc()),
  is_blocked=False,
  explorer_url='https://explorer.devnet.grid.tf/api/v1'
)
```

Reply:
if we used to get with a not defined instance, it won't create it 
![image](https://user-images.githubusercontent.com/46320202/101607441-49ad1d80-3a0d-11eb-83d2-88863a19d2e8.png)

we need the `get` behavior

Reply:
the correct method you need is `find`. `get` fails to create a new instance, which is bug in the new method as it requires `owner_tname` and `flavor` which may be solved